internal/runners/languages/languages.go

```diff
@@ -7,38 +7,55 @@ import (
 	"github.com/ActiveState/cli/internal/output"
 	"github.com/ActiveState/cli/pkg/platform/model"
 	"github.com/ActiveState/cli/pkg/project"
+	"github.com/go-openapi/strfmt"
 )
 
+// Languages manages the listing execution context.
 type Languages struct {
 	out     output.Outputer
 	project *project.Project
 }
 
+// NewLanguages prepares a list execution context for use.
 func NewLanguages(prime primeable) *Languages {
 	return &Languages{
 		prime.Output(),
 		prime.Project(),
 	}
 }
 
+// Listing represents the output data of a list of languages.
 type Listing struct {
 	Languages []model.Language `json:"languages"`
 }
 
+// MarshalOutput implements the output.Marshaller interface.
 func (l Listing) MarshalOutput(f output.Format) interface{} {
 	if f == output.PlainFormatName {
 		return l.Languages
 	}
 	return l
 }
 
+// Run executes the list behavior.
 func (l *Languages) Run() error {
 	if l.project == nil {
 		return locale.NewInputError("err_no_project")
 	}
-	langs, err := model.FetchLanguagesForCommit(l.project.CommitUUID())
+
+	commitUUID, err := usableCommitUUID(l.project)
 	if err != nil {
-		return err
+		return locale.WrapError(
+			err, "err_no_usable_commitid", "Cannot obtain a usable commit id",
+		)
+	}
+
+	langs, fail := model.FetchLanguagesForCommit(commitUUID)
+	if fail != nil {
+		return locale.WrapError(
+			fail, "err_fetching_languages",
+			"Cannot obtain languages for commit id {{.V0}}", commitUUID.String(),
+		)
```

Comment:
Why did this need to be added?

Reply:
In other PRs wrapping at the handler level (i.e the run func) has been emphasized. It also clarifies which commit id was used which errored out.